src/prefect/orion/schemas/responses.py

```diff
@@ -129,3 +140,76 @@ class Config:
     worker_pool_id: UUID
     worker_pool_queue_id: UUID
     flow_run: schemas.core.FlowRun
+
+
+@copy_model_fields
+class FlowRunResponse(ORMBaseModel):
+
+    name: str = FieldFrom(schemas.core.FlowRun)
+    flow_id: UUID = FieldFrom(schemas.core.FlowRun)
+    state_id: Optional[UUID] = FieldFrom(schemas.core.FlowRun)
+    deployment_id: Optional[UUID] = FieldFrom(schemas.core.FlowRun)
+    work_queue_name: Optional[str] = FieldFrom(schemas.core.FlowRun)
+    flow_version: Optional[str] = FieldFrom(schemas.core.FlowRun)
+    parameters: dict = FieldFrom(schemas.core.FlowRun)
+    idempotency_key: Optional[str] = FieldFrom(schemas.core.FlowRun)
+    context: dict = FieldFrom(schemas.core.FlowRun)
+    empirical_policy: FlowRunPolicy = FieldFrom(schemas.core.FlowRun)
+    tags: List[str] = FieldFrom(schemas.core.FlowRun)
+    parent_task_run_id: Optional[UUID] = FieldFrom(schemas.core.FlowRun)
+    state_type: Optional[schemas.states.StateType] = FieldFrom(schemas.core.FlowRun)
+    state_name: Optional[str] = FieldFrom(schemas.core.FlowRun)
+    run_count: int = FieldFrom(schemas.core.FlowRun)
+    expected_start_time: Optional[DateTimeTZ] = FieldFrom(schemas.core.FlowRun)
+    next_scheduled_start_time: Optional[DateTimeTZ] = FieldFrom(schemas.core.FlowRun)
+    start_time: Optional[DateTimeTZ] = FieldFrom(schemas.core.FlowRun)
+    end_time: Optional[DateTimeTZ] = FieldFrom(schemas.core.FlowRun)
+    total_run_time: datetime.timedelta = FieldFrom(schemas.core.FlowRun)
+    estimated_run_time: datetime.timedelta = FieldFrom(schemas.core.FlowRun)
+    estimated_start_time_delta: datetime.timedelta = FieldFrom(schemas.core.FlowRun)
+    auto_scheduled: bool = FieldFrom(schemas.core.FlowRun)
+    infrastructure_document_id: Optional[UUID] = FieldFrom(schemas.core.FlowRun)
+    infrastructure_pid: Optional[str] = FieldFrom(schemas.core.FlowRun)
+    created_by: Optional[CreatedBy] = FieldFrom(schemas.core.FlowRun)
+    worker_pool_name: Optional[str] = Field(
+        default=None,
+        description="The name of the flow run's worker pool.",
+        example="my-worker-pool",
+    )
+    worker_pool_queue_name: Optional[str] = Field(
+        default=None,
+        description="The name of the flow run's worker pool queue.",
+        example="my-worker-pool-queue",
+    )
+    state: Optional[schemas.states.State] = FieldFrom(schemas.core.FlowRun)
+
+    @classmethod
+    async def from_orm_model(
```

Comment:
I'm a concerned about performance here. This is adding two database queries to one of our most frequently called endpoints.

1. Is this implementation going to issue queries 1 by 1 for each flow run? E.g. for 200 flow runs does this issue another 400 SELECT statements?
2. Is it essential to include this data in the first place?

Reply:
1. Yes. The alternative I see is to alter the flow run query to also retrieve worker pool and worker pool information via a join. Would you expect the performance be better in with that approach?
2. I believe so. This was added so that the worker pool name and worker pool queue name can be display for flow runs in the UI. The alternative is to return the worker pool queue ID and then the UI makes API calls to retrieve the name of the pool and queue. That would mean two additional API calls for each flow run unless updates are made to the existing routes.

Reply:
@zangell44 I have implemented a different approach that might be more palatable. I added necessary relationships so additional explicit queries for worker pool and worker pool queue information isn't necessary. Let me know if you think this approach will be better for performance.